pwiz_tools/Skyline/Alerts/AlertDlg.cs

```diff
@@ -181,11 +182,31 @@ private void btnMoreInfo_Click(object sender, EventArgs e)
             }
         }
 
-        public void AddMessageBoxButtons(MessageBoxButtons messageBoxButtons) 
+        public void AddMessageBoxButtons(MessageBoxButtons messageBoxButtons, MessageBoxDefaultButton defaultButton)
         {
+            var buttons = new List<Button>();
             foreach (var dialogResult in GetDialogResults(messageBoxButtons).Reverse())
             {
-                AddButton(dialogResult);
+                buttons.Add(AddButton(dialogResult));
+            }
+
+            Button focusButton = null;
+            switch (defaultButton)
+            {
+                case MessageBoxDefaultButton.Button1:
+                    focusButton = buttons.LastOrDefault();
+                    break;
+                case MessageBoxDefaultButton.Button2:
+                    focusButton = buttons.Count == 3 ? buttons[1] : buttons.Count == 2 ? buttons[0] : null;
+                    break;
+                case MessageBoxDefaultButton.Button3:
+                    focusButton = buttons.FirstOrDefault();
+                    break;
+            }
+
+            if (focusButton != null)
+            {
+                focusButton.Select();

```

Comment:
Should we be setting the "AcceptButton" property on the form instead?
The AcceptButton is the button that gets pressed if the user presses Enter while the focus is on some other control such as a textbox. That's the standard way of designating the "Default" button on a form, but it might not make a difference on AlertDlg since AlertDlg only has buttons on it.

Reply:
Done, thanks.